webapi/src/main/scala/org/knora/webapi/responders/v2/SearchResponderV2.scala

```diff
@@ -117,6 +113,57 @@ trait SearchResponderV2 {
       r <- gravsearchCountV2(q, user)
     } yield r
 
+  /**
+   * Performs a Gravsearchquery to find resources that link to the specified resource.
+   *
+   * @param resourceIri the IRI of the resource to which incoming links are to be found.
+   * @param offset      the offset to be used for paging.
+   * @param rendering   the schema of the response.
+   * @param user        the client making the request.
+   */
+  def searchIncomingLinksV2(
+    resourceIri: IRI,
+    offset: Int,
+    rendering: SchemaRendering,
+    user: User,
+  ): Task[ReadResourcesSequenceV2] = {
+    val query: String =
+      s"""
+         |PREFIX knora-api: <http://api.knora.org/ontology/knora-api/simple/v2#>
+         |
+         |CONSTRUCT {
+         |?incomingRes knora-api:isMainResource true .
+         |
+         |?incomingRes ?incomingProp <$resourceIri> .
+         |
+         |} WHERE {
+         |
+         |?incomingRes a knora-api:Resource .
+         |
+         |?incomingRes ?incomingProp <$resourceIri> .
+         |
+         |<$resourceIri> a knora-api:Resource .
+         |
+         |?incomingProp knora-api:objectType knora-api:Resource .
+         |
+         |knora-api:isRegionOf knora-api:objectType knora-api:Resource .
+         |knora-api:isPartOf knora-api:objectType knora-api:Resource .
+         |
+         |FILTER NOT EXISTS {
+         |?incomingRes  knora-api:isRegionOf <$resourceIri> .
+         |}
+         |
+         |FILTER NOT EXISTS {
+         |?incomingRes  knora-api:isPartOf <$resourceIri> .
+         |?incomingRes knora-api:seqnum ?seqnum .
+         |}
+         |
+         |} OFFSET $offset
+         |""".stripMargin
+
+    gravsearchV2(query, rendering, user)
+  }
+
```

Comment:
suggestion: In order to keep the abstraction levels intact, I would move the implementation of this route to the trait implementation, rather than implementing it directly in the trait.